File: scala/digitalwallet/app/src/main/scala/core/domain/model/ProcessTransactionRequest.scala
package core.domain.model

import core.domain.enums.SubwalletType.SubwalletType
import core.domain.enums.TransactionType.TransactionType

case class ProcessTransactionRequest(
  amount: BigDecimal,
  batchId: Option[String] = None,
  idempotencyKey: String,
  originatorWalletId: String,
  originatorSubwalletType: SubwalletType,
  beneficiaryWalletId: Option[String] = None,
  beneficiarySubwalletType: Option[SubwalletType] = None,
  transactionType: TransactionType,
  metadata: Option[TransactionMetadata] = None,
)

File: kotlin/digital-wallet/src/main/kotlin/digitalwallet/core/domain/models/ProcessTransactionRequest.kt
package digitalwallet.core.domain.models

import digitalwallet.core.domain.enums.SubwalletType
import digitalwallet.core.domain.enums.TransactionType
import java.math.BigDecimal

data class ProcessTransactionRequest(
    val amount: BigDecimal,
    val batchId: String? = null,
    val idempotencyKey: String,
    val originatorWalletId: String,
    val originatorSubwalletType: SubwalletType,
    val beneficiaryWalletId: String? = null,
    val beneficiarySubwalletType: SubwalletType? = null,
    val type: TransactionType,
    val metadata: TransactionMetadata? = null
)